File: C++/PrimeRecursion.cpp
#include <iostream>
#include <vector>
#include <sys/resource.h>

std::vector<int> checkPrime(int x) {
if (x > 2) {
        std::vector<int> primes = checkPrime(x-1);
        for (int value:primes) {
            if(x % value == 0)
                return primes;
        }
        primes.push_back(x);
        return primes;
    }
    std::vector<int> temp (1,2);
    return(temp);
}


int main() {
    
    for (int value:checkPrime(60000)) {
        printf("%d, ", value);
    }
    return 0;
}

File: Python/PrimeRecursion.py
# This is a sample Python script.

# Press ⌃R to execute it or replace it with your code.
# Press Double ⇧ to search everywhere for classes, files, tool windows, actions, and settings.
import sys
sys.setrecursionlimit(0x100000)

def checkPrime(x):
    if x > 2:
        primes = checkPrime(x-1)
        for value in primes:
            if x % value == 0:
                return primes
        primes.append(x)
        return primes
    return([2])

print(checkPrime(30000))
